### PR TITLE
Leave one channel out

### PR DESCRIPTION
**Pull request by MikeLippincott**
This is the leave-one-channel out analysis for the Elastic net that predicts cytokine abundance from morphology features.

### CODE REVIEW

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+print(test_set_all.shape, test_set_75.shape, test_set_50.shape)
+
+
+# In[9]:
+
+
+# get the train test splits from each group
+# 100% test set
+test_set_all
```
Comment (gwaybio) [inline 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:120]: will this print out a lot of stuff? maybe not worth doing?

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+# In[10]:
+
+
+# combine all testing sets together while preserving the index
+testing_data_set = pd.concat(
+    [test_set_all, testing_data_set_75, testing_data_set_50], axis=0
+)
+testing_data_set = testing_data_set.sort_index()
+testing_data_set
```
Comment (gwaybio) [inline 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:163]: probably best to print just the `.head()`?

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+# combine all training sets together while preserving the index
+training_data_set = pd.concat([training_data_set_75, training_data_set_50], axis=0)
```
Comment (gwaybio) [inline 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:166]: I'm a bit confused - why are you combining these training datasets? Won't this result in duplicated samples?

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        save_path = pathlib.Path(f"../indexes/{cell_type}/{MODEL_TYPE}")
+    elif nomic == False:
+        save_path = pathlib.Path(f"../indexes/{cell_type}/{MODEL_TYPE}")
+elif aggregation == False:
+    if nomic == True:
+        save_path = pathlib.Path(f"../indexes/{cell_type}/{MODEL_TYPE}")
+    elif nomic == False:
+        save_path = pathlib.Path(f"../indexes/{cell_type}/{MODEL_TYPE}")
+else:
+    print("Error")
```
Comment (gwaybio) [inline 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:208]: all of the `save_path` variables are the same so why have all of this logic?

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+        )
+    elif nomic == False:
+        index_data.to_csv(f"{save_path}/aggregated_sc_data_split_indexes.tsv", sep="\t")
+elif aggregation == False:
+    if nomic == True:
+        index_data.to_csv(f"{save_path}/sc_and_nomic_data_split_indexes.tsv", sep="\t")
+    elif nomic == False:
+        index_data.to_csv(f"{save_path}/sc_split_indexes.tsv", sep="\t")
+else:
+    print("Error")
```
Comment (gwaybio) [inline 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:232]: this is a non-specific error message that is not informative if anyone gets to this

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/2.LOCO_data_split.py [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+# In[8]:
+
+
+# create the combinations of features
+# out of 5 channels, we can have 0, 1, 2, 3, 4, or 5 channels
+# even with 0 channels, we still have the non channel features
+# these are areshape features
+# set the feature combination lists
+dict_of_feature_combinations = {
```
Comment (gwaybio) [inline 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/2.LOCO_data_split.py:120]: why are there so many combinations? Shouldn't there just be five combinations? Maybe this is why the analysis is taking so long?

File: 11.bulk_Morphology_Elastic_Network_LOCO/README.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# Morphological features Predict Secretome Protein Abundance via Elastic Net
```
Comment (gwaybio) [inline 11.bulk_Morphology_Elastic_Network_LOCO/README.md:1]: this is a very helpful document!

**Review by gwaybio — APPROVED** [review]
a few, relatively minor comments. The biggest comment is why there are 32 channel combinations - this seems more complicated than needed

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/results/feature_combinations_PBMC.toml
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/results/feature_combinations_PBMC.toml:1]: renamed file to be more descriptive.

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py (hunk 1 above)
Comment (MikeLippincott) [reply 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:120]: Yes, I removed this line

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py (hunk 2 above)
Comment (MikeLippincott) [reply 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:163]: removed print statements all together

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py (hunk 3 above)
Comment (MikeLippincott) [reply 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:166]: Good point, I removed these

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py (hunk 4 above)
Comment (MikeLippincott) [reply 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:208]: Good point, I removed these

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py (hunk 5 above)
Comment (MikeLippincott) [reply 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:232]: Made this error message more specific

File: 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py (hunk 3 above)
Comment (MikeLippincott) [reply 11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py:166]: I train on one complete training set but the 50/50 vs 75/25 training/testing are split out first. I double checked and there are no duplicates.

File: .gitignore [hunk 8]
```diff
@@ -38,6 +39,8 @@ __pycache__/
 # index files
 4.sc_Morphology_Neural_Network_MLP_Model/0.hyperparameter_optimization/indexes/*
 6.bulk_Morphology_Elastic_Network/0.split_data/indexes/*
+11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/indexes/*
+11.bulk_Morphology_Elastic_Network_LOCO/1.train_models/models/*
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply .gitignore:43]: I will add these to a figshare instead of github.

[the remark this answers is not in the document]
Comment (MikeLippincott) [reply .gitignore:43]: Uploading to github

**Pull request merged**